spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, :type => :controller do
+
+  describe "#facebook" do
+
+    context "given user does persist" do
+      before do
+        OmniAuth.config.mock_auth[:do_omniauth] = FactoryGirl.create(:auth_hash, :facebook)
+
+        request.env["devise.mapping"] = Devise.mappings[:user]
+        request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:do_omniauth]
+      end
+
+      it "should successfully authenticate using Facebook authentication hash and display a success message in flash" do
+        get :facebook
+        user = User.find_by(email: 'testuser@facebook.com')
+        expect(user.id).not_to eq(nil)
+        expect(flash[:notice]).to eq("Successfully authenticated from Facebook account.")
+      end
+    end
+
+    context "given user does not persist" do
+      before do
+        OmniAuth.config.mock_auth[:do_omniauth] = FactoryGirl.create(:auth_hash, :facebook, :does_not_persist)
+
+        request.env["devise.mapping"] = Devise.mappings[:user]
+        request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:do_omniauth]
+      end
```

Comment:
You're absolutely right. I tried it again and it worked, which makes total sense. Not sure what I did wrong when I tried it last night. I probably commented out the wrong line of code or something. Who knows, it was pretty late 😴